nsxt/resource_nsxt_transport_node.go

```diff
@@ -0,0 +1,1987 @@
+				"vmk_install_migration": {
+					Type:        schema.TypeList,
+					Description: "The vmknic and logical switch mappings",
+					Optional:    true,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"destination_network": {
+								Type:        schema.TypeString,
+								Required:    true,
+								Description: "The network id to which the ESX vmk interface will be migrated",
+							},
+							"device_name": {
+								Type:        schema.TypeString,
+								Required:    true,
+								Description: "ESX vmk interface name",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func getTransportZoneEndpointSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeList,
+		Optional:    true,
+		Description: "Transport zone endpoints",
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"transport_zone_id": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "Unique ID identifying the transport zone for this endpoint",
+				},
+				"transport_zone_profile_id": {
+					Type:        schema.TypeList,
+					Optional:    true,
+					Description: "Identifiers of the transport zone profiles associated with this transport zone endpoint on this transport node",
+					Elem: &schema.Schema{
+						Type: schema.TypeString,
+					},
+				},
+			},
+		},
+	}
+}
+
+func getUplinksSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeList,
+		Optional:    true,
+		Description: "Uplink/LAG of VMware vSphere Distributed Switch connected to the HostSwitch",
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"uplink_name": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "Uplink name from UplinkHostSwitch profile",
+				},
+				"vds_lag_name": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					Description: "Link Aggregation Group (LAG) name of Virtual Distributed Switch",
+				},
+				"vds_uplink_name": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					Description: "Uplink name of VMware vSphere Distributed Switch (VDS)",
+				},
+			},
+		},
+	}
+}
+
+func getHostSwitchProfileIDsSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeList,
+		Description: "Identifiers of host switch profiles to be associated with this host switch",
+		Optional:    true,
+		Elem: &schema.Schema{
+			Type: schema.TypeString,
+		},
+	}
+}
+
+func getIPAssignmentSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeList,
+		Description: "Specification for IPs to be used with host switch virtual tunnel endpoints",
+		MaxItems:    1,
+		Required:    true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"assigned_by_dhcp": {
+					Type:        schema.TypeBool,
+					Optional:    true,
+					Description: "Enables DHCP assignment. Should be set to true",
+				},
+				"static_ip_list": {
```

Comment:
If this is single entry, should we call this `list`?

Reply:
I followed the API naming. However this assignment mode allows multiple IPs (ip_addresses), so technically the name reflects using multiple IPs, right?

Reply:
I see what you mean, still I would suggest not to use `list` because its not in line with naming conventions that is used so far in the provider. So `static_ip` or `static_ips` would be more consistent.